lib/src/main/java/com/cloudbees/groovy/cps/DepthTrackingEnv.java

```diff
@@ -5,8 +5,10 @@
  */
 public interface DepthTrackingEnv extends Env {
 
-    /** Limit on how deeply environments can recurse */
-    int MAX_LEGAL_DEPTH = 3000;
+    /** Limit on how deeply environments can recurse.
+     *  Capped somewhat low to try to limit the ability to run a program that will generate a StackOverflowError when serialized. */
+    int MAX_LEGAL_DEPTH = 1024;
```

Comment:
I am afraid this might kill legitimate programs which just happen to use, say, a recursive function to process a list or whatever. (Yes common Groovy idioms would typically make that unnecessary, but…)

Reply:
yes, but necessary

Reply:
They'd run into a StackOverflowError eventually if they *weren't* NonCPS code though -- we're just setting an explicit threshold.  1024 is high enough that I feel comfortable saying "if you need more than this, you shouldn't be running it as part of CPS execution." 

There's always the escape valve of NonCPS annotations and ability to rewrite logic to use a non-recursive form.